airflow/api_connexion/openapi/v1.yaml

```diff
@@ -1447,8 +1447,9 @@ components:
           readOnly: true
           nullable: true
         state:
-          $ref: '#/components/schemas/DagState'
           readOnly: True
+          oneOf:
+            - $ref: '#/components/schemas/DagState'
```

Comment:
am i correct that `readOnly: True` was ignored as sibling element of `$ref` without this change?

Reply:
So that was the idea in the beginning since spectral started complaining about that (having siblings of $ref). The [docs](https://swagger.io/docs/specification/using-ref/) refer that siblings are ignored, but it seems that the code generation isn't prepared for this. 

I'm still a bit unfamiliar with the project, do you think this should be reverted?

Reply:
Based on the docs you linked, i think the change you introduced looks correct, but I agree this is better handled as a separate follow issue.